Scripts/application.py

```diff
@@ -14,99 +16,216 @@ def hello_world():
+                raw_data: Tuple = cursor.fetchone()
+                data: List = [
+                    {
+                        "person_ID": raw_data[0],
+                        "email": raw_data[1],
+                        "assignment ID": raw_data[2]}]
                 json_data = json.dumps(data)
             except:
-                return('There was an error. Check that the requested ID is valid.')
+                return(
+                    'There was an error. Check that the requested ID is valid.')
         return json_data
     
 @app.route('/api/v1/forbidden_pairing', methods = ["GET","POST"])
 def return_forbidden_pairings():
     
     if request.method == "POST":
-        new_giver_constraint = request.get_json()["santa"]
-        new_receiver_constraint = request.get_json()["assignment"]
+    
+        # verify that request contains a valid request  
+        try:
+            request_json = request.get_json()
+        except:
+            abort(
+                400,
+                description = "Please verify that request contains valid json.")
+        try:
+            new_giver_constraint = request_json['santa_ID']
+        except:
+            abort(
+                400,
+                description = "Please verify that request contains a "
+                              "'santa_ID' key.")
+        try:
+            new_receiver_constraint = request_json['assignment_ID']
+        except:
+            abort(
+                400,
+                description = "Please verify that request contains an "
+                              "'assignment_ID' key.")
+        
+        # add new constraint to database
         with sqlite3.connect(path_to_db) as connection:
             cursor = connection.cursor()
-            command = 'SELECT (person_ID) FROM people WHERE email = ?;'
+            command = 'SELECT (email) FROM people WHERE person_ID = ?;'
             cursor.execute(command, (new_giver_constraint,))
             giver_ID = cursor.fetchone()[0]
-            cursor.execute(command, (new_receiver_constraint,))
+            cursor.execute(command, (new_receiver_constraint,)) # finish swapping IDs for emails!!!
             receiver_ID = cursor.fetchone()[0]
-            command = 'INSERT into forbiddenPairings (santa, santa_ID, assignment, assignment_ID) VALUES (?, ?, ?, ?);'
-            cursor.execute(command, (new_giver_constraint, giver_ID, new_receiver_constraint, receiver_ID))            
+            command = ('INSERT into forbiddenPairings (santa, santa_ID '
+                       'assignment, assignment_ID) VALUES (?, ?, ?, ?);')
+            cursor.execute(
+                command,
+                (
+                    new_giver_constraint,
+                    giver_ID,
+                    new_receiver_constraint,
+                    receiver_ID))
+             
+            #return posted record
+            cursor.execute('SELECT last_insert_rowid()')

```

Comment:
similar to the above comment about specifying the fields you want

Reply:
ok